### DISCUSSION TITLE
Physics Experiment: Variations to jumping and falling

### DISCUSSION THREAD

**Discussion by qreeves** [redeclipse/base#1128] [Maps and Content Showcase]
UPDATE: You now only fall faster when holding crouch, so it is more controllable.

As an experiment, I have introduced two new variables to the master branch, `gravityfall` and `gravityjump`. These two variables modify the force of gravity depending on the situation:

* `gravityfall` increases gravity after you've reached the peak of your jump, making you fall faster. It is currently set to 1.5x.
* `gravityjump` decreases gravity while you're holding the jump button until you reach the peak. It is currently set to 0.75x.

Related commit: https://github.com/red-eclipse/base/commit/a4b18333df03e1a7fb4f9efb14edc9c0a776d67d

The idea is to maintain current capabilities while reducing the "floaty" feeling while in the air. You may need to hold jump to cover the same distances as before, and the variables may need to be more finely tuned than just basic math. Give it a try, play with these variables if you want, and let me know what you think.

Further information:
* http://www.mathforgameprogrammers.com/gdc2016/GDC2016_Pittman_Kyle_BuildingABetterJump.pdf
* https://www.youtube.com/watch?v=hG9SzQxaCm8

A visualisation of the jumping arc change:
![arcs](https://user-images.githubusercontent.com/3297929/34723044-e68a7050-f594-11e7-9765-f232055c7d32.png)

**Comment by cloudziq** [comment]
As for experiment, Its very good addition. Very subtle change, almost unnoticeable. My type is /gravityjump 0.6 (a little higher when keeping jump pressed), /gravityfall is ok for me. :)

**Comment by SniperGoth** [comment]
Seems like the height you can get while clawing (claw + holding space ) using these new physics has increased quite a bit, from my tests.

EDIT: A approximate visual comparison on Battlefield, from 1.5's claw 
![20180119204339](https://user-images.githubusercontent.com/24359960/35175229-d4abdc08-fd59-11e7-9a4b-cd78a68c7b02.png)
 And now, 1.9.9's physics with the claw 
![20180119204521](https://user-images.githubusercontent.com/24359960/35175254-eba15046-fd59-11e7-86da-f71cbabfdb7e.png)

Im liking how it is so far...

**Comment by qreeves** [comment]
UPDATE: You now only fall faster when holding crouch, so it is more controllable.
